### PR TITLE
Update Bakta & Fix errors with Bakta and AMRFinderPlus DB

### PR DESCRIPTION
**Pull request by bebatut**
FOR CONTRIBUTOR:
* [X] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] License permits unrestricted use (educational + commercial)
* [X] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

Hi,

This PR updates Bakta and (hopefully) fixes some errors that we got with Bakta:
- Update in DM the ARMFinderPlus DB provided in recent version of Bakta DB (otherwise Bakta fails)
- Use versioned AMRFinderPlus data table (need to merge #6034 first)
- Add filter on ARMFinderPlus DB version

Bérénice

### CODE REVIEW

File: tools/bakta/test-data/V5.1_light_2024-01-19/amrfinderplus-db/2024-05-02.2/AMRProt.phr
Comment (bgruening) [inline tools/bakta/test-data/V5.1_light_2024-01-19/amrfinderplus-db/2024-05-02.2/AMRProt.phr:1]: This one seems to be too big.

File: tools/bakta/test-data/V5.1_light_2024-01-19/amrfinderplus-db/2024-05-02.2/AMRProt.psq
Comment (bgruening) [inline tools/bakta/test-data/V5.1_light_2024-01-19/amrfinderplus-db/2024-05-02.2/AMRProt.psq:1]: This one is even bigger.

File: tools/bakta/test-data/V5.1_light_2024-01-19/amrfinderplus-db/2024-05-02.2/AMRProt.phr
Comment (bebatut) [reply tools/bakta/test-data/V5.1_light_2024-01-19/amrfinderplus-db/2024-05-02.2/AMRProt.phr:1]: I know. I just can not find out how to reduce it...

**Comment by bebatut** [comment]
To do in the PR
- [ ] Remove AMRFinderPlus DB in Bakta DM
- [ ] Create a small Bakta DB for tests in Bakta DM that will be stored on Zenodo
- [ ] Create a small AMRFinderPlus DB:
   - [ ] Download BDD from NCBI ftp
   - [ ] Downsample FASTA files
   - [ ] Run AMRFinderPlus DM to build indexes and other files

File: data_managers/data_manager_build_bakta_database/data_manager/bakta_build_database.py [hunk 1]
```diff
@@ -64,6 +62,7 @@ def fetch_db_versions(self):
             self.DB_VERSIONS_URL = self.DB_TEST_URL
         try:
             with requests.get(self.DB_VERSIONS_URL) as resp:
+                print(resp.content)
```
Comment (bgruening) [inline data_managers/data_manager_build_bakta_database/data_manager/bakta_build_database.py:65]:
```suggestion
```

File: tools/bakta/bakta.xml [hunk 2]
```diff
@@ -1,111 +1,109 @@
...
+            LOG file
+======================================*#
+| tee '$logfile'
         ]]></command>
     <inputs>
       <!-- DB and file INPUT -->
         <section name="input_option" title="Input/Output options" expanded="true">
-            <param name="bakta_db_select" type="select" label="The bakta database">
+            <param name="input_file" type="data" format="fasta,fasta.gz" label="Select genome in fasta format"/>
+            <param name="min_contig_length" type="integer" optional="true" min="0" label="Minimum contig size" help="Minimum contig size (default = 1; 200 in compliant mode) (--min-contig-length)"/>
```
Comment (bgruening) [inline tools/bakta/bakta.xml:94]:
```suggestion
            <param argument="--min-contig-length" type="integer" optional="true" min="0" label="Minimum contig size" help="Minimum contig size (default = 1; 200 in compliant mode)"/>
```

File: tools/bakta/bakta.xml [hunk 3]
```diff
@@ -171,82 +169,141 @@
...
             <section name="output_files">
                 <param name="output_selection" value="file_tsv,file_gff3,file_gbff,file_embl,file_fna,file_ffn,file_faa,hypo_tsv,hypo_fa,sum_txt,file_json,file_plot,log_txt"/>
             </section>
-            <output name="annotation_tsv" value="TEST_1/TEST_1.tsv" lines_diff="2"/>
-            <output name="annotation_gff3" value="TEST_1/TEST_1.gff3" lines_diff="2"/>
-            <output name="annotation_gbff" value="TEST_1/TEST_1.gbff" lines_diff="8"/>
-            <output name="annotation_embl" value="TEST_1/TEST_1.embl" lines_diff="6"/>
-            <output name="annotation_fna" value="TEST_1/TEST_1.fna"/>
-            <output name="annotation_ffn" value="TEST_1/TEST_1.ffn"/>
-            <output name="annotation_faa" value="TEST_1/TEST_1.faa"/>
```
Comment (bgruening) [inline tools/bakta/bakta.xml:234]: can those "TEST_1/TEST_1.faa" files be deleted now?

Comment (bebatut) [reply tools/bakta/bakta.xml:234]: I removed them all

File: tools/bakta/bakta.xml (hunk 2 above)
Comment (bebatut) [reply tools/bakta/bakta.xml]: Thanks for the catch. I replaced it there and somewhere else

File: data_managers/data_manager_build_bakta_database/data_manager/bakta_build_database.py (hunk 1 above)
Comment (bebatut) [reply data_managers/data_manager_build_bakta_database/data_manager/bakta_build_database.py]: Done in extra commit. Thanks

**Review by clsiguret — COMMENTED** [review]
This looks good for me :+1: 
Thanks for doing it :grinning:

**Review by clsiguret — APPROVED** [review]

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks :-)

**Pull request merged**